theano/sandbox/rng_mrg.py

```diff
@@ -1388,6 +1391,7 @@ def binomial(self, size=None, n=1, p=0.5, ndim=None, dtype='int64',
                  nstreams=None):
         # TODO : need description for method, parameter and return
         if n == 1:
+            p = undefined_grad(p)
```

Comment:
I think you need to call `as_tensor_variable` on `p` before that, as `undefined_grad` expect a `Variable`, but in some cases `p` is a Python `float`, so tests are failing.